### PR TITLE
757/germanriano/mosel

### PR DESCRIPTION
**Pull request by jcmuel**
I deleted the test file that had an incompatible license and created a new test file. Moreover, I also added several keywords and types.

### CODE REVIEW

File: pygments/lexers/mosel.py [hunk 1]
```diff
@@ -27,123 +27,400 @@ class MoselLexer(RegexLexer):
...
-            (r'(integer|is_binary|real)', Keyword.Type),
-            (r'(\+|\-|\*|/|=|<=|>=|==|\||\^|<|>|<>|\.\.|:=|:)', Operator),
+            (words((
+                'range', 'array', 'set', 'list', 'mpvar', 'mpproblem', 'linctr', 'nlctr', 'integer',
+                'string', 'real', 'boolean', 'text', 'time', 'date', 'datetime', 'returned', 'Model',
+                'Mosel', 'counter', 'xmldoc',
+                'is_sos1', 'is_sos2', 'is_integer', 'is_binary', 'is_continuous',
+                'is_free', 'is_semcont', 'is_semint', 'is_partint'
+                )), Keyword.Type),
+            (r'(\+|\-|\*|/|=|<=|>=|==|\||\^|<|>|<>|\.\.|\.|:=|:|::|in|mod|div)', Operator),
```
Comment (birkenfeld) [inline pygments/lexers/mosel.py]: `:` and `::` need to swap places to have `::` recognized.

File: pygments/lexers/mosel.py [hunk 2]
```diff
@@ -27,123 +27,400 @@ class MoselLexer(RegexLexer):
...
             (r'[()\[\]{},;:]+', Punctuation),
```
Comment (birkenfeld) [inline pygments/lexers/mosel.py]: Is `:` still needed here as punctuation?

Comment (jcmuel) [reply pygments/lexers/mosel.py]: You are right. It's not needed. I just removed it.

File: pygments/lexers/mosel.py (hunk 1 above)
Comment (jcmuel) [reply pygments/lexers/mosel.py]: Thanks for the hint. I just reordered the `:` and `::`.

**Comment by birkenfeld** [comment]
Thanks!

**Pull request merged**